hack/presubmit-tests.sh

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Documentation about this script and how to use it can be found
+# at https://github.com/knative/hack
+
+export DISABLE_MD_LINTING=1
+export DISABLE_MD_LINK_CHECK=1
+export PRESUBMIT_TEST_FAIL_FAST=1
+
+export KNATIVE_SERVING_VERSION=${KNATIVE_SERVING_VERSION:-latest}
+export KNATIVE_EVENTING_VERSION=${KNATIVE_EVENTING_VERSION:-latest}
+source $(dirname $0)/../vendor/knative.dev/hack/presubmit-tests.sh
+
+function build_tests() {
+  local failed=0
+  header "Building code"
+  make build || failed=1
+  if (( failed )); then
+    results_banner "Build failed"
+    exit ${failed}
+  fi
+
+  # Use a docker image to run the verify-codegen.sh script so that it's always running in the same environment.
+  # header "Check auto-generated code"
+  docker run -it --rm -v$(pwd):/src/ golang sh -c 'cd /src && ./hack/verify-codegen.sh' || failed=2
+  if (( failed )); then
+    results_banner "Code generation failed"
+    exit ${failed}
+  fi
+
+  # Check that we don't have any forbidden licenses in our images.
+  subheader "Checking for forbidden licenses"
+  report_build_test Check_Licenses check_licenses || failed=3
+  if (( failed )); then
+    results_banner "License check failed"
+    exit ${failed}
+  fi
+}
+
+function unit_tests() {
+  local failed=0
+  header "Unit tests for $(go_mod_module_name)"
+  make test || failed=1
+  if (( failed )); then
+    results_banner "Unit tests failed"
+    exit ${failed}
+  fi
+
+  header "Running built-in template tests"
+  make test-templates || failed=2
+  if (( failed )); then
+    results_banner "Template tests failed"
+    exit ${failed}
+  fi
+}
+
+function integration_tests() {
+  local failed=0
+  header "Running integration tests"
+  make test-integration || failed=1
```

Comment:
Where will this run? Integration tests requires some set up (e.g. docker daemon, cluster, registry).